examples/cholesky.plv

```diff
@@ -0,0 +1,103 @@
+import prelude;
+
+-- "Matrix Computations" 4th ed., Golub and Van Loan.  Algorithm
+-- 4.1.2, the LDL^T decomposition for a symmetric, positive-definite
+-- matrix.
+--
+-- On success, the argument A now contains the lower-triangular factor
+-- L and the diagonal vector D.
+ldlt {m}
+  (inout A :: double[m, m])
+  :: s8 := (
+
+  v :: double[m];
+
+  for j in 0:m -> (
+    -- Compute v[1:j].
+    for i in 0:j -> (
+       v[i] <- A[j, i] * A[i, i];
+    );
+    v[j] <- A[j, j] - A[j, 0:j] * v[0:j];
+
+    -- Store d[j].
+    A[j, j] <- v[j];
+
+    -- Compute L[j+1:n, j].
+    A[j+1:m, j] <- (A[j+1:m, j] - A[j+1:m, 0:j]*v[0:j]) / v[j];
+  );
+
+  return 0;
+);
+
+-- "Advanced Kalman Filtering, Least-Squares and Modeling.", Bruce
+-- P. Gibbs.  Algorithm 10.2-2, the UDU^T decomposition for a
+-- symmetric, positive-definite matrix.
+--
+-- On exit, the argument A now contains the upper-triangular factor U
+-- and the diagonal vector D.
+udut {m}
+  (inout A :: double[m, m])
+  :: Void := (
+  for j in m-1:0:-1 -> (  -- column loop
+    alpha := if (A[j, j] > 0.0) then (1.0 / A[j, j]) else 0.0;
+    for k in 0:j -> (     -- row loop
+      beta := A[k, j];
+      A[k, j] <- alpha*beta;  -- scale elements in j column
+      A[0..k, k] <- A[0..k, k] - beta*A[0..k, j];  -- modify column k
+    );
+  );
+);
+
+
+-- "Matrix Computations" 4th ed., Golub and Van Loan.  Algorithm
+-- 4.2.2, the Cholesky decomposition for a symmetric,
+-- positive-definite matrix.
+--
+-- On exit, the argument A now contains the lower-triangular
+-- Cholesky factor G.
+cholesky {m}
+  (inout A :: double[m, m])
+  :: Void := (
+  for k in 0:m -> (
+    sq := sqrt(A[k, k]);
+    A[k, k] <- sq;
+    A[k+1:m, k] <- A[k+1:m, k] / sq;
+    for j in k+1:m -> (
+      A[j:m, j] <- A[j:m, j] - A[j:m, k]*A[j, k];
```

Comment:
with updates like this, you need to be careful because plover currently will not warn you if you are aliasing memory badly (if the rhs computation depends on values already updated on the lhs). the lines you've written in this PR all look ok.